### PR TITLE
Populate config.json from EC2 metadata

### PR DESCRIPTION
**Pull request by ab77**
* Hackathon 2k20 project

Change-type: patch

### CODE REVIEW

File: src/metadata.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import * as constants from './lib/constants';
```
Comment (CameronDiver) [inline src/metadata.ts:1]: This filename (metadata.ts) needs to be more specific.

File: src/metadata.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import * as request from './lib/request';
+import { readFileSync, writeFileSync } from 'fs';
```
Comment (CameronDiver) [inline src/metadata.ts:3]: Don't ever use `Sync` functions in the supervisor, use the promise based ones:

```suggestion
import { promises as fs } from 'fs';
```

and then 
`fs.readFile`, fs.writeFile`

File: src/metadata.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import { reboot } from './device-state';
+import log from './lib/supervisor-console';
+
+export const initialized = (async () => {
+	if (constants.configJsonNonAtomicPath != null) {
+		const configJsonPath = constants.configJsonNonAtomicPath;
```
Comment (CameronDiver) [inline src/metadata.ts:9]: This variable is useless

File: src/metadata.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+		if (
+			!config.hasOwnProperty('unmanaged') &&
+			!config.hasOwnProperty('registered_at') &&
+			!config.hasOwnProperty('applicationId') &&
+			!config.hasOwnProperty('deviceId')
+		) {
+			const opts: request.requestLib.CoreOptions = {};
+
+			log.debug('Attempting to initialise from cloud metadata');
+			const url = 'http://169.254.169.254/latest/user-data';
```
Comment (CameronDiver) [inline src/metadata.ts:20]: What is this IP?

**Review by CameronDiver — CHANGES_REQUESTED** [review]
Can I get a little more information about what this PR is for please? It should be in the PR description, but as it stands I don't know what this is for, and more specifically I don't know that it should be added to the supervisor

File: src/metadata.ts (hunk 4 above)
Comment (ab77) [reply src/metadata.ts:20]: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-add-user-data.html

**Pull request closed without merging**